spec/system/nested_documentable_spec.rb

```diff
@@ -0,0 +1,314 @@
+
+      fill_in_required_fields(factory, path)
+
+      %w[clippy empty logo].take(documentable.class.max_documents_allowed).each do |filename|
+        click_link "Add new document"
+
+        attach_file "Choose document", file_fixture("#{filename}.pdf")
+
+        within all(".document-fields").last do
+          expect(page).to have_css ".loading-bar.complete"
+        end
+      end
+      click_button submit_button_text
+
+      if factory != :dashboard_action
+        within "#documents" do
+          expect(page).to have_content "Documents (3)"
+          expect(page).to have_link text: "empty.pdf"
+          expect(page).to have_css "a[rel=nofollow]", text: "empty.pdf"
+          expect(page).to have_link text: "clippy.pdf"
+          expect(page).to have_css "a[rel=nofollow]", text: "clippy.pdf"
+          expect(page).to have_link text: "logo.pdf"
+          expect(page).to have_css "a[rel=nofollow]", text: "logo.pdf"
+          expect(page).not_to have_css "a[target=_blank]"
+        end
+      end
+      expect(page).to have_content notice_text
+    end
+
+    describe "Metadata" do
+      let(:factory) { (factories - [:dashboard_action]).sample }
+
+      scenario "download document without metadata" do
+        do_login_for(user, management: management_section?(path))
+        visit path
+
+        fill_in_required_fields(factory, path)
+        documentable_attach_new_file(file_fixture("logo_with_metadata.pdf"))
+        click_button submit_button_text
+
+        expect(page).to have_content notice_text
+
+        io = URI.parse(find_link(text: "PDF")[:href]).open
+        reader = PDF::Reader.new(io)
+
+        expect(reader.info[:Keywords]).not_to eq "Test Metadata"
+        expect(reader.info[:Author]).not_to eq "Test Developer"
+        expect(reader.info[:Title]).not_to eq "logo_with_metadata.pdf"
+        expect(reader.info[:Producer]).not_to eq "Test Producer"
+        expect(reader.info).to eq({})
+      end
+    end
+
+    describe "When allow attached documents setting is disabled" do
+      before do
+        Setting["feature.allow_attached_documents"] = false
+        create(:document, documentable: documentable)
+      end
+
+      scenario "Add new document button and documents list should not be available" do
+        do_login_for(user, management: management_section?(path))
+        visit path
+
+        expect(page).not_to have_content("Add new document")
+
+        click_button submit_button_text
+
+        expect(page).not_to have_css("#documents")
+      end
+    end
+  end
+
+  describe "Only for edit path" do
+    let!(:proposal) { create(:proposal, author: user) }
+
+    scenario "Should show persisted documents and remove nested_field" do
+      create(:document, documentable: proposal)
+      login_as user
+      visit edit_proposal_path(proposal)
+
+      expect(page).to have_css ".document-fields", count: 1
+
+      click_link "Remove document"
+
+      expect(page).not_to have_css ".document-fields"
```

Comment:
What do you think about adding the `#new_document_link` expectations here, as part of the «Remove redundant tests from "Only for edit path" describe» commit? That means we'd need to add `Setting["uploads.documents.max_amount"] = 1` to this test or move it to a new `before` block at the beginning of the file.

```suggestion
      visit edit_proposal_path(proposal)

      expect(page).to have_css ".document-fields", count: 1
      expect(page).not_to have_link "Add new document"

      click_link "Remove document"

      expect(page).not_to have_css ".document-fields"
      expect(page).to have_link "Add new document"
```